src/codewhisperer/service/recommendationHandler.ts

```diff
@@ -222,9 +225,8 @@ export class RecommendationHandler {
             startTime = performance.now()
             this.lastInvocationTime = startTime
             const mappedReq = runtimeLanguageContext.mapToRuntimeLanguage(request)
-            const codewhispererPromise = pagination
-                ? client.listRecommendations(mappedReq)
-                : client.generateRecommendations(mappedReq)
+            const codewhispererPromise =
+                pagination && !isSM ? client.listRecommendations(mappedReq) : client.generateRecommendations(mappedReq)
```

Comment:
Why isn't this special-case needed here for cloud9? There are already various callsites such as [this](https://github.com/aws/aws-toolkit-vscode/blob/756ee6913a88d2a8fa4c0cbc3f46d4dd12703873/src/codewhisperer/commands/invokeRecommendation.ts#L68-L76) which set `pagination` depending on `isCloud9`. So it seems suspicious that this special-case is being added at this low level.

Reply:
Cloud9 is special in that it uses the `generateRecommendations` endpoint (for IAM auth) but can't handle the pagination, due to UI limitations. So we are creating a path that allows for pagination and also IAM auth. All other paginated code paths with use `listRecommendations` and auth tokens.

Note: on the linked code it is sending `false` for pagination, since it is Cloud9.